z3.cabal

```diff
@@ -59,6 +59,7 @@ Library
     -- Ban to mtl-2.1: modify in MonadState causes <<loop>> in mtl-2.1
     Build-depends:       base >= 4.5 && <5,
                          containers,
+                         extra,
```

Comment:
Could you create an internal module `Z3.Common` and add your own version of `whenMaybe` there?